packages/core/src/types/builder-options.ts

```diff
@@ -175,14 +180,14 @@ export type InterfaceTypeOptions<
 export type EnumTypeOptions<
   Types extends SchemaTypes,
   Param extends EnumParam,
-  Values extends EnumValues<Types>,
+  Values extends EnumValues<Types, Param>,
```

Comment:
this would be a breaking change, and will break other packages (I think you already mentioned this in the PR description)